### PR TITLE
Closes #3202 Index.inferred_type

### LINKED ISSUE #3202: Index.inferred_type
Add the property inferred_type to enable Index.inferred type to match pandas:
https://pandas.pydata.org/docs/reference/api/pandas.Index.inferred_type.html

```python
In [5]: c = pd.Categorical(["a","b","a","c"])

In [6]: i = pd.Index(c)

In [7]: i.inferred_type
Out[7]: 'categorical'
```

### PR DESCRIPTION
**Pull request by ajpotts**
Closes #3202 Index.inferred_type

Adds the property `inferred_type`.

### CODE REVIEW

File: arkouda/pdarrayclass.py [hunk 1]
```diff
@@ -399,6 +395,18 @@ def max_bits(self, max_bits):
...
+        """
+        Whether pdarrays are the same size and all entries are equal.
+        """
+        if isinstance(other, pdarray):
+            if other.size != self.size:
+                return False
+            else:
+                return all(self == other)
+        else:
+            return False
```
Comment (stress-tess) [inline arkouda/pdarrayclass.py:408]: it seems code from https://github.com/Bears-R-Us/arkouda/pull/3209 has slipped into this PR

**Review by stress-tess — APPROVED** [review]
Looks great! only comment is I believe some code from a different PR found it's way into this one. It doesn't really matter since it's small and I've approved them both, but it is something to be careful about in general

**Review by jaketrookman — APPROVED** [review]

File: arkouda/pdarrayclass.py (hunk 1 above)
Comment (ajpotts) [reply arkouda/pdarrayclass.py]: Sorry, I guess I was rushing this one a bit.  I did remove it.

Comment (stress-tess) [reply arkouda/pdarrayclass.py]: no worries!!

**Review by stress-tess — APPROVED** [review]

**Pull request merged**